### PR TITLE
[Plot] - Adding getAllSelections() API

### LINKED ISSUE #1545: API for getting selection out of Plot.Scatter
Similar to `getAllBars()` on `Plot.Bar`.

### PR DESCRIPTION
**Pull request by bluong**
Close #1545.

### CODE REVIEW

File: src/components/plots/areaPlot.ts [hunk 1]
```diff
@@ -90,6 +90,10 @@ export module Plot {
       attrToProjector["stroke"] = attrToProjector["stroke"] || d3.functor(this._defaultFillColor);
       return attrToProjector;
     }
+
+    protected _getSelector(): string {
+      return ".area";
```
Comment (jtlan) [inline src/components/plots/areaPlot.ts]: This will fail to return the line part of the area?

**Comment by jtlan** [comment]
Can we get the selection method to coordinate with the `Drawer`s somehow? Right now, the drawers sometimes attach classes to the selection they draw, but this implementation adds a separate `_getSelector()` API point on `Plot`, which means the `Drawer` and `Plot` could easily get out of sync because we have the information in two places.

File: src/components/plots/areaPlot.ts (hunk 1 above)
Comment (bluong) [inline src/components/plots/areaPlot.ts]: Wait I thought we didn't want to return the line part

**Comment by bluong** [comment]
@jtlan I deliberately didn't want to borrow from the Drawer since right now the Drawer's architecture seems very black-boxy and complex, meaning that we may want to change it in the future.  In which case, I wanted this code to be neutral to whatever change we have in Drawers.

I am ok with reverting to the Drawer usage but slightly hesitant

**Comment by jtlan** [comment]
@bluong I meant more that `_getSelector()` could be implemented on the drawer, allowing the Plot to get the selection the drawer is using without duplicating logic.

File: src/components/plots/abstractPlot.ts [hunk 2]
```diff
@@ -419,6 +419,22 @@ export module Plot {
...
+          allSelections[0].push(selection);
+        });
+      });
+
+      return allSelections;
+    }
+
+    protected _getSelector(): string {
+      return "";
+    }
```
Comment (jtlan) [inline src/components/plots/abstractPlot.ts]: probably should remove this?

**Comment by jtlan** [comment]
Nice, the `Drawer`-based implementation is much more compact. You left something from the previous implementation in **abstractPlot.ts** though.

**Comment by jtlan** [comment]
@crmorford For testing, maybe create a small test where you try to set the CSS class (using `classed()`) or stylings (using `style()`) of the returned `D3.Selection`.

**Comment by crmorford** [comment]
Could you guys leave me a small example when you add new features / add stuff to the API? I can figure it out this time, but I think it's more efficient that way.

**Comment by crmorford** [comment]
*Works on all standard Bar/Area/Line/Scatter Plots, Clustered Bar, Stacked Bar, Stacked Area, Pie Chart
*Doesn't give unexpected behavior when called on axes/labels/empty plots/gridlines/legend
*Speed better than initial render on large amounts of data

Nice! GIM

**Pull request merged**